packages/platform-sdk-ada/src/services/client.ts

```diff
@@ -35,7 +36,16 @@ export class ClientService implements Contracts.ClientService {
 	}
 
 	public async transactions(query: Contracts.ClientTransactionsInput): Promise<Coins.TransactionDataCollection> {
-		throw new Exceptions.NotImplemented(this.constructor.name, "transactions");
+		const transactions: object[] = (await this.get(`v2/wallets/${query.address}/transactions`)) as object[];
```

Comment:
@faustbrian I've got another question here. The `cardano-wallet` api always requires a `walletId`. What field should I use out of those in the contract `ClientTransactionsInput`? I used address, but it's not right. Actually none of them seem to be right.

Reply:
What's wrong with the `address` property?

Reply:
Well, technically it's not an address. Address look different, they start with `addr_`. This is a "walletId", I think it's something that could be derived from the root key pair.

Reply:
Do we get this ID from the API when we request the wallet information? If so then we could store this property as meta data on the wallet instance and retrieve that when needed.

Reply:
Yeah..
Wallet Id is from the Wallet API.. different than address.
I never could track down exactly how it’s derived.

Reply:
To the wallet api we submit only our public key, and it responds with the wallet id. So it must be derivable from there. I'm trying to figure this out.
```
POST http://localhost:8090/v2/wallets
Content-Type: application/json

{
  "name": "Shelley publicKey-only Wallet",
  "account_public_key": "52f79be32364ecd0846652a749a876e4eb550889273404991a92007ac88773f2a21d47deda83a49cb435a2adfd0fb6945efc5b07f384db06d27c69cc4f586805",
  "address_pool_gap": 20
}
```
And we get the reponse:
```
{
  "address_pool_gap": 20,
  "state": {
    "status": "syncing",
    "progress": {
      "quantity": 0,
      "unit": "percent"
    }
  },
  "balance": {
    "reward": {
      "quantity": 0,
      "unit": "lovelace"
    },
    "total": {
      "quantity": 0,
      "unit": "lovelace"
    },
    "available": {
      "quantity": 0,
      "unit": "lovelace"
    }
  },
  "name": "Shelley publicKey-only Wallet",
  "delegation": {
    "next": [],
    "active": {
      "status": "not_delegating"
    }
  },
  "id": "82d63879d8914699da73edbb65b091da16e21964",
  "tip": {
    "height": {
      "quantity": 0,
      "unit": "block"
    },
    "time": "2019-07-24T20:20:16Z",
    "epoch_number": 0,
    "absolute_slot_number": 0,
    "slot_number": 0
  },
  "assets": {
    "total": [],
    "available": []
  }
}
```

Reply:
![Screenshot from 2021-02-10 08-31-26](https://user-images.githubusercontent.com/959563/107504695-764ef400-6b7a-11eb-83e1-097882a2aa72.png)
@faustbrian This is a screenshot from Daedalus. This is how they show transactions. They have senders addresses and receipients addresses. Both can be just one or can be many. How about we use the first of the senders as `sender()`, the first of recipients as `recipient()` and then all recipients as `recipients()`? This way we would only probably miss showing some of the extra senders, but we still show one address that belongs to the sender. If there are more, they should still belong to the same account.

Reply:
That should work. We will link to the explorer anyways for full details.